### DISCUSSION TITLE
How to initialize a sharedValue to change from/to value?

### DISCUSSION THREAD

**Discussion by t6adev** [software-mansion/react-native-reanimated#2236] [Q&A]
Hi there,
In my case, I'd like to change from/to the value of sharedValue by any trigger like in useEffect at the same time.

Currently, it looks my code works, like the below.
```
const sharedValue = useSharedValue(from);
// e.g from/to is changed in 0/100 or 100/0
useEffect(() => {
  sharedValue.value = from;
  sharedValue.value = withTiming(to);
}, [from, to]);
```

Is this the correct way?
Thanks.

**Comment by Noitidart** [comment]
This is interesting, that code shouldn't work it seems. As I and another user her have this problem - https://github.com/software-mansion/react-native-reanimated/discussions/2643
